Lib/test/test_asyncio/test_futures2.py

```diff
@@ -1,13 +1,68 @@
 # IsolatedAsyncioTestCase based tests
 import asyncio
+import inspect
 import unittest
+import traceback
 
 
 def tearDownModule():
     asyncio.set_event_loop_policy(None)
 
 
 class FutureTests(unittest.IsolatedAsyncioTestCase):
+    maxDiff = None
+
+    async def test_future_traceback(self):
```

Comment:
We can have a simple test that awaits in a loop and checks that the traceback is always the same (the example from the BPO, essentially).

We can also have this test, where each await is from a different place, but as I mentioned in https://github.com/python/cpython/pull/30274/files#r830857535 it doesn't need all this complexity.

Reply:
Test is simplified now.